src/app/accessible-table/accessible-table.component.ts

```diff
@@ -0,0 +1,221 @@
+    }
+
+  }
+
+  private updateFirstRowHeight(): void {
+    this.firstRowHeight = this.elementRef
+    .nativeElement
+    .parentElement
+      ?.querySelector('tbody')
+      ?.offsetHeight
+      || this.firstRowHeight
+  }
+
+  onAllCheckboxChange({checked}) {
+    if (checked) {
+      this.selectedRowsChange.emit(this.rows)
+    } else {
+      this.selectedRowsChange.emit([])
+    }
+  }
+
+  // Change gets called before click.
+  onCheckboxChange(event, row, index) {
+    this.onRowClick(event, row, index)
+  }
+
+  onCheckboxClick(event, row, index) {
+    // We don not want to trigger the row click when clicking on checkbox.
+    event.stopPropagation()
+  }
+
+  rangeSelect(to, check) {
+    let from = this.rows.indexOf(this.lastCheckedRow)
+
+    // When nothing is selected yet.
+    if (from === -1) return this.oneSelect(to, check)
+
+    return this.rangeSelectFrom(from, to, check)
+  }
+
+  oneSelect(index, check) {
+    this.rangeSelectFrom(index, index, check)
+  }
+
+  rangeSelectFrom(from: number, to: number, check: boolean) {
+      const left = Math.min(from, to)
+      const right = Math.max(from, to)
+      const clone = [...this.selectedRows]
+
+      for (let i = left; i <= right; i++) {
+        clone[i] = check ? this.rows[i] : null
+      }
+
+      this.selectedRowsChange.emit(clone)
+      this.lastCheckedRow = this.rows[to]
+  }
+
+  onRowClick(event, row, index) {
+    if (this.shiftKey) {
+      return this.rangeSelect(index, !this.selectedRows[index])
+    }
+
+    if (this.ctrlKey || this.metaKey) {
+      return this.oneSelect(index, !this.selectedRows[index])
+    }
+
+    this.selectedRowChange.emit(row)
+    this.lastCheckedRow = row
+  }
+
+  onRowKeydown(event, row, index) {
+    // Only work on Enter and space.
+    if (event.key !== 'Enter') return;
+
+    return this.onRowClick(event, row, index)
+  }
+
+  selectedSet() {
+    const selected = new Set(this.selectedRows)
+
+    selected.delete(null)
+    selected.delete(undefined)
+
+    return selected;
+  }
+
+  emitSelect() {
+    return
+  }
+
+  get isIndeterminate() {
+    // Has both selected and unselected items. Might be better to use a
+    let hasSelected = false;
+    let hasUnselected = false;
+
+    return this.rows.find((_value, index) => {
+      const v = this.selectedRows[index]
+
+      hasSelected = hasSelected || Boolean(v);
+      hasUnselected = hasUnselected || Boolean(!v);
+
+      return (hasSelected && hasUnselected)
+    })
+  }
+
+  get allCheckboxIsChecked() {
+    return this.selectedRows.find(x => x != null)
+  }
+
+  onDragStart(event, row, index) {
+    const emptyImage = new Image();
+    event.dataTransfer?.setDragImage(emptyImage, 0, 0); // Set an empty image
+
+    const isSelected = this.selectedRows[index]
+
+    const selected = isSelected
+      ? this.selectedSet()
+      : new Set([row]);
+
+    this.selectionDragStarted.emit({
+      event,
+      selected,
+    })
+  }
```

Comment:
I would have liked to decouple this from the table. It would be nice to have a better abstraction to work with draggable.

Maybe the complete table API is strange. Since when are tables concerned about selectedRows? Might it not be more correct to decouple that too by offering a better templating API. I have ideas but I also need to draw a line and say it's good enough.